### PR TITLE
JAMES-2246 Integration tests for SMTP configuration

### PR DESCRIPTION
**Pull request by chibenwa**
Reopen of #1168

### CODE REVIEW

File: server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * <p>If the sender was not authenticated, and the sender not part of an authorized network (as defined in smtpserver.xml)
+ * it will not match.</p>
+ *
+ * <pre><code>
+ * &lt;mailet match=&quot;IsSmtpRelayAllowed&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * </code></pre>
+ */
+public class IsSmtpRelayAllowed extends GenericMatcher {
+    public Collection<MailAddress> match(Mail mail) {
+        return mail.getRecipients().stream()
```
Comment (mbaechler) [inline server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java]: you don't use the stream, don't you ?

Comment (chibenwa) [reply server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java]: No I did not.

Updated to a if.

File: server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpAuthorizedAddresses.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+import com.google.common.base.Charsets;
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.Duration;
+import com.jayway.awaitility.core.ConditionFactory;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.builder.RequestSpecBuilder;
+import com.jayway.restassured.http.ContentType;
+
+public class SmtpAuthorizedAddresses {
```
Comment (rouazana) [inline server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpAuthorizedAddresses.java]: missing "Test"?

File: server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * <p>Matches mails that are allowed to be relayed by the SMTP server. This includes authenticated users and authorized networks.</p>
+ * <p>If the sender was not authenticated, and the sender not part of an authorized network (as defined in smtpserver.xml)
+ * it will not match.</p>
+ *
+ * <pre><code>
+ * &lt;mailet match=&quot;IsSmtpRelayAllowed&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * </code></pre>
+ */
+public class IsSmtpRelayAllowed extends GenericMatcher {
```
Comment (rouazana) [inline server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java]: what is the purpose of this matcher which is only combining SMTPIsAuthNetwork and SMTPAuthSuccessful?

Comment (chibenwa) [reply server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java]: Ok, I discovered `SMTPIsAuthNetwork`...

Comment (chibenwa) [reply server/protocols/protocols-smtp/src/main/java/org/apache/james/transport/matchers/IsSmtpRelayAllowed.java]: I removed IsSmtpRelayAllowed and edited confs accordingly :-)

**Comment by chibenwa** [comment]
test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpAuthorizedAddressesTest.java [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        createJamesServer(SmtpConfiguration.builder()
+            .requireAuthentication()
+            .withAutorizedAddresses("172.0.0.0/8"));
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG)) {
+
+            messageSender.sendMessage(FROM, TO);
+
+            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpAuthorizedAddressesTest.java]: still a 10 second sleep ?

File: server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpBracketEnforcement.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    public void recipientWithNoBracketsShouldBeRejectedWhenBracketRequired() throws Exception {
+        createJamesServer(SmtpConfiguration.builder()
+            .requireBracketEnforcement());
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.authentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG, USER, PASSWORD)) {
+
+            messageSender.sendMessageNoBracket(USER, USER);
+
+            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpBracketEnforcement.java]: still ?

File: server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpIdentityVerification.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        createJamesServer(SmtpConfiguration.builder()
+            .requireAuthentication()
+            .verifyIdentity());
+
+        try (SMTPMessageSender messageSender =
+                 SMTPMessageSender.authentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG, ATTACKER, ATTACKER_PASSWORD)) {
+
+            messageSender.sendMessage(USER, USER);
+
+            Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpIdentityVerification.java]: another one

**Comment by chibenwa** [comment]
Handled on the other PR

File: server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpAuthorizedAddresses.java (hunk 2 above)
Comment (rouazana) [reply server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpAuthorizedAddresses.java]: in fact that concerns all your test files in this PR

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**